File: examples/fib_loop.js
let fib = (n) => {
    let a = 0;
    let b = 1;
    let temp = 0;
    if (n == 0) return a;
    for (let c = 0; c < n ; c++) {
        temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

let r = fib(30);
console.log(r);

File: examples/fib_loop.rs
fn fib(n: i32) -> i32 {
    let mut a = 0;
    let mut b = 1;
    let mut temp;

    for _ in 0..n {
        temp = a + b;
        a = b;
        b = temp;
    }

    b
}

fn main() {
    // assert_eq!(fib(0), 1);
    // assert_eq!(fib(1), 1);
    // assert_eq!(fib(2), 2);
    // assert_eq!(fib(3), 3);
    // assert_eq!(fib(4), 5);
    dbg!(fib(35));
}
